aptos-move/aptos-e2e-comparison-testing/src/execution.rs

```diff
@@ -320,95 +318,66 @@ impl Execution {
-                        features.enable(FeatureFlag::VM_BINARY_FORMAT_V7);
-                    } else {
-                        features.enable(FeatureFlag::VM_BINARY_FORMAT_V6);
-                    }
-                };
-                if let Some(debugger) = debugger_opt {
-                    let data_view =
-                        DataStateView::new(debugger, version, executor.data_store().clone());
-                    let mut features =
-                        Features::fetch_config(&data_view.as_move_resolver()).unwrap_or_default();
-                    enable_v7(&mut features);
-                    return Some(executor.try_exec_entry_with_state_view(
-                        senders,
-                        entry_function,
-                        &data_view.as_move_resolver(),
-                        features,
-                    ));
-                } else {
-                    let mut features =
-                        Features::fetch_config(&executor.data_store().clone().as_move_resolver())
-                            .unwrap_or_default();
-                    enable_v7(&mut features);
-                    return Some(executor.try_exec_entry_with_state_view(
-                        senders,
-                        entry_function,
-                        &executor.data_store().clone().as_move_resolver(),
-                        features,
-                    ));
-                }
-            }
+    ) -> Result<(WriteSet, Vec<ContractEvent>), VMStatus> {
+        // Always add Aptos (0x1) packages.
+        add_aptos_packages_to_data_store(&mut state, compiled_package_cache);
+
+        // Add other modules.
+        if package_info.is_compilable() {
+            add_packages_to_data_store(&mut state, package_info, compiled_package_cache);
         }
+
+        // Update features if needed to the correct binary format used by V2 compiler.
+        let mut features = Features::fetch_config(&state).unwrap_or_default();
+        if v2_flag {
+            features.enable(FeatureFlag::VM_BINARY_FORMAT_V7);
+        } else {
+            features.enable(FeatureFlag::VM_BINARY_FORMAT_V6);
+        }
+        state.set_features(features);
+
+        // We use executor only to get access to block executor and avoid some of
+        // the initializations, but ignore its internal state, i.e., FakeDataStore.
+        let executor = FakeExecutor::no_genesis();
+        let txns = vec![txn.clone()];
+
         if let Some(debugger) = debugger_opt {
-            let data_view = DataStateView::new(debugger, version, executor.data_store().clone());
-            Some(
-                executor
-                    .execute_transaction_block_with_state_view([txn.clone()].to_vec(), &data_view)
-                    .map(|res| res[0].clone().into()),
-            )
+            let data_view = DataStateView::new(debugger, version, state);
+            executor
+                .execute_transaction_block_with_state_view(txns, &data_view)
```

Comment:
hm inside this function (execute_transaction_block_with_state_view), we access "self.get_state_view()" on line:
`trace_map.0 = Self::trace(trace_data_dir.as_path(), self.get_state_view());`
should that take the view passed in?

Reply:
Good point! Fixed